tools/autograd/templates/Functions.cpp

```diff
@@ -1861,6 +1861,14 @@ std::tuple<Tensor, Tensor, Tensor> _trilinear_backward(const Tensor& grad_out, c
   return std::tuple<Tensor, Tensor, Tensor>(grad_i1, grad_i2, grad_i3);
 }
 
+Tensor log1p_backward(const Tensor& grad, const Tensor& self) {
+  if (self.is_sparse()) {
+    AT_ERROR("log1p of a sparse tensor is made to be non-differentiable since ",
+      "local gradient of zero is 1 / (0 + 1) = 1 and it makes the tensor dense");
```

Comment:
Since I've given one comment about an error message, here's another: maybe we should add some guidance like "use a different mathematical operation which preserves sparsity of gradients, or report a bug if you think this is in error"